common/blockchain_util.py

```diff
@@ -40,7 +41,8 @@ def read_contract_address(self, net_id, path, key):
         return Web3.toChecksumAddress(contract[str(net_id)][key])
 
     def contract_instance(self, contract_abi, address):
-        return self.web3_object.eth.contract(abi=contract_abi, address=address)
+        web3_object = Web3(self.provider)
```

Comment:
@prashantramangupta why this change?

Reply:
because while reusing the web3 connection, sometimes it loses connection with `infura`.
So there is no direct way to know when the connection got expired.
The right way is to set up a new connection on exception and retry with a new connection.
But it is time-consuming and needs to test other services.